src/Microsoft.AspNetCore.Mvc.Core/Filters/FilterCollection.cs

```diff
@@ -86,6 +87,17 @@ public IFilterMetadata Add(Type filterType, int order)
                 throw new ArgumentException(message, nameof(filterType));
             }
 
+            if (typeof(IFilterFactory).IsAssignableFrom(filterType))
```

Comment:
This does not cover all possible scenarios, but it covers the issue and minimally affects compat.